### ISSUE TITLE
Field "Select": Different values in Gutenberg preview (array) / Website preview (string)

### ISSUE THREAD

**Issue by camya** [getblocklab/block-lab#181]
(I found this issue in the **no-mock-attributes branch**. #180 - Not sure, if it's also in the develop branch.)

Looks like there is a small problem with the return values of the select fields.

I get different data via "Gutenberg preview" (array) and "Website preview" (string).

### Steps to Reproduce

1. Create a publisher "select" field. The choices are:

```
pub-a : Publisher A
pub-b : Publisher B
pub-c : Publisher C
```

2. Create PHP template with this code:

```
<?php
echo "publisher: ";
var_dump(block_value( 'publisher' ));
?>

```
3. Select the first publisher from the dropdown.

4. Output in "Gutenberg preview":

`publisher: array(1) { [0]=> string(5) "pub-a"}`

5. Output in "Website preview":

`publisher: string(5) „pub-a“`

**Expected behavior:** 

string(5) „pub-a“ should be the output in both, Gutenberg and Website preview.

**Comment by lukecarbis** [comment]
Note that the expected behaviour _should be_ that a select returns an array every time (not a string), so that multiselect is always supported.

**Comment by camya** [comment]
I'm not sure about returning an array all the time.

The Block-Lab Developer designs the form and specify, if a select field is single or multi. Also he creates the template with his design decision in mind.

I guess, "single select" fields are used more frequently than multi select ones. Handling an array value in the template every time makes templating more complicated than handling a simple string.

**My suggestion:** 

- Return **single select** as STRING
- Return **multi select** as ARRAY

**Example: Template output for single select returned as STRING**
```
<?php if (block_value( 'publisher' )): ?>
  <p>
     You selected "<?php echo esc_html(block_value( 'publisher' )); ?>".
  </p>
<?php endif; ?>
```

**Example: Template output for single and multi select returned as an ARRAY**

If you always return an array, also if no multi select is used, this makes it harder to write templates.

```
<?php $publisherId = block_value( 'publisher' ); ?>
<?php if (isset($publisherId[0]) && $publisherId[0]): ?>
  <p>
    You selected "<?php echo esc_html($publisherId[0]); ?>".
  </p>
<?php endif; ?>
```

**Comment by lukecarbis** [comment]
Good point.

Maybe it makes most sense to have Select and Multiselect as different input types.

OR we could use Multicheckbox instead (overriding the current single checkbox).

Any thoughts @RobStino ?

**Comment by RobStino** [comment]
@greenphp makes a good point about simplifying for single select.
I think we should take the single>string, multi>array approach for both the select field as well as the checkbox field (noting that there is no current support for multi-checkboxes).
The question now is:
- Do we have a single field type that offers an option that "switches" between single and multi (this would effect templating), or
- Offer seperate field types for each.

**Comment by lukecarbis** [comment]
I think we offer separate field types, to reduce confusion. 

The other questions are:
- Do we offer a multiselect and a multicheckbox (what’s the use case for needing both?)
- If we have a multicheckbox and a toggle field, do we need a single checkbox?

**Comment by RobStino** [comment]
**Do we offer a multiselect and a multicheckbox (what’s the use case for needing both?)**
Use case is multicheckbox is good for a short list. Multiselect is good for a long list I.e. Countries
**If we have a multicheckbox and a toggle field, do we need a single checkbox?**
Perhaps we remove single checkbox. If you really want a single checkbox for UX, you can just do a single option on a multicheckbox.

**Comment by lukecarbis** [comment]
Resolved in #185.
